src/settings/ExpenseClassSettings/ExpenseClassSettings.js

```diff
@@ -50,6 +51,11 @@ const ExpenseClassSettings = ({ stripes }) => {
       visibleFields={visibleFields}
       rowFilter={<ExpenseClassHelper />}
       validate={validate}
+      canCreate={hasEditPerms}
+      actionSuppressor={{
+        edit: () => !hasEditPerms,
+        delete: () => !hasEditPerms,
```

Comment:
you should frame the const hasEditPerms based on the specific PUT permission.
make another declaration for hasDeletePerms based on the specific DELET permission.
hook them accordingly to the actionSuppressor